pkg/apis/ceph.rook.io/v1/scc.go

```diff
@@ -38,6 +37,7 @@ func NewSecurityContextConstraints(name, namespace string) *secv1.SecurityContex
 			Namespace: namespace,
 		},
 		AllowPrivilegedContainer: true,
+		AllowHostDirVolumePlugin: true,
```

Comment:
Why do we need to enable this?

Reply:
mon canary pods won't come up if we don't enable this enabled.  Rook logs show : 
```
ceph-cluster-controller: failed to get ceph daemons versions, this typically happens during the first cluster initialization. failed to run 'ceph versions'. . unable to get monitor info from DNS SRV with service name: ceph-mon2021-11-08T11:19:26.938+0000 7f9fb7fff700 -1 failed for service _ceph-mon._tcp
```

Reply:
I don't understand how the error relates to this setting. Isn't this an env issue?

Reply:
> I don't understand how the error relates to this setting. Isn't this an env issue?

All of the ceph daemons have a hostPath volume for the logs, right? So the hostPath volume is needed or else all the daemon pods would fail to start. This setting also matches what we already have in the [operator-openshift.yaml](https://github.com/rook/rook/blob/master/cluster/examples/kubernetes/ceph/operator-openshift.yaml#L14).

Reply:
Ah right, I always think of flex when I see this I don't know why. Makes sense.